llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -7529,6 +7535,43 @@ SDValue AArch64TargetLowering::LowerINIT_TRAMPOLINE(SDValue Op,
                      EndOfTrmp);
 }
 
+SDValue AArch64TargetLowering::LowerFMUL(SDValue Op, SelectionDAG &DAG) const {
+  EVT VT = Op.getValueType();
+  auto &Subtarget = DAG.getSubtarget<AArch64Subtarget>();
+  if (VT.getScalarType() != MVT::bf16 ||
+      !(Subtarget.hasBF16() && (Subtarget.hasSVE() || Subtarget.hasSME())))
```

Comment:
Is it possible to make this more of an explicit opt-in to match the setOperationAction side. For example:
```
if (VT.getScalarType() != MVT::bf16 ||
    (Subtarget->hasSVEB16B16() && Subtarget->isNonStreamingSVEorSME2Available())
  return LowerToPredicatedOp

assert(Subtarget.hasBF16() && ....


```